File: backend/src/main/java/com/foodorder/service/MenuService.java
package com.foodorder.service;

import com.foodorder.model.MenuItem;
import com.foodorder.repository.MenuRepository;

import java.sql.SQLException;
import java.util.List;

public class MenuService {

    private MenuRepository menuRepository;

    public MenuService() {
        this.menuRepository = new MenuRepository();
    }

    public String[] addItem(MenuItem item) {
        String[] result = new String[2];
        if (item.getName() == null || item.getName().trim().isEmpty()) {
            result[0] = "error"; result[1] = "Item name is required"; return result;
        }
        if (item.getPrice() <= 0) {
            result[0] = "error"; result[1] = "Price must be greater than zero"; return result;
        }
        if (item.getCategoryId() <= 0) {
            result[0] = "error"; result[1] = "Please select a valid category"; return result;
        }
        try {
            boolean saved = menuRepository.save(item);
            result[0] = saved ? "success" : "error";
            result[1] = saved ? "Menu item added successfully" : "Failed to add menu item";
        } catch (SQLException e) {
            result[0] = "error"; result[1] = "Database error: " + e.getMessage();
        }
        return result;
    }

    public List<MenuItem> getAllItems() {
        try {
            return menuRepository.findAll();
        } catch (SQLException e) {
            e.printStackTrace(); return null;
        }
    }

    public MenuItem getItemById(int id) {
        if (id <= 0) return null;
        try {
            return menuRepository.findById(id);
        } catch (SQLException e) {
            e.printStackTrace(); return null;
        }
    }

    public List<MenuItem> getItemsByCategory(int categoryId) {
        if (categoryId <= 0) return null;
        try {
            return menuRepository.findByCategory(categoryId);
        } catch (SQLException e) {
            e.printStackTrace(); return null;
        }
    }

    public String[] updateItem(MenuItem item) {
        String[] result = new String[2];
        if (item.getId() <= 0) {
            result[0] = "error"; result[1] = "Invalid item ID"; return result;
        }
        if (item.getName() == null || item.getName().trim().isEmpty()) {
            result[0] = "error"; result[1] = "Item name is required"; return result;
        }
        if (item.getPrice() <= 0) {
            result[0] = "error"; result[1] = "Price must be greater than zero"; return result;
        }
        try {
            MenuItem existing = menuRepository.findById(item.getId());
            if (existing == null) {
                result[0] = "error"; result[1] = "Menu item not found"; return result;
            }
            boolean updated = menuRepository.update(item);
            result[0] = updated ? "success" : "error";
            result[1] = updated ? "Menu item updated successfully" : "Failed to update menu item";
        } catch (SQLException e) {
            result[0] = "error"; result[1] = "Database error: " + e.getMessage();
        }
        return result;
    }

    public String[] toggleAvailability(int id, boolean status) {
        String[] result = new String[2];
        if (id <= 0) {
            result[0] = "error"; result[1] = "Invalid item ID"; return result;
        }
        try {
            boolean toggled = menuRepository.toggleAvailability(id, status);
            result[0] = toggled ? "success" : "error";
            result[1] = toggled ? (status ? "Item marked as available" : "Item marked as unavailable") : "Item not found";
        } catch (SQLException e) {
            result[0] = "error"; result[1] = "Database error: " + e.getMessage();
        }
        return result;
    }

    public String[] deleteItem(int id) {
        String[] result = new String[2];
        if (id <= 0) {
            result[0] = "error"; result[1] = "Invalid item ID"; return result;
        }
        try {
            MenuItem existing = menuRepository.findById(id);
            if (existing == null) {
                result[0] = "error"; result[1] = "Menu item not found"; return result;
            }
            boolean deleted = menuRepository.delete(id);
            result[0] = deleted ? "success" : "error";
            result[1] = deleted ? "Menu item deleted successfully" : "Failed to delete menu item";
        } catch (SQLException e) {
            result[0] = "error"; result[1] = "Database error: " + e.getMessage();
        }
        return result;
    }
}

File: frontend/src/services/menuService.js
import api from './api';

// GET /api/menu/all
// returns list of all menu items
export const getAllItems = async () => {
    try {
        const response = await api.get('/menu/all');
        return response.data;
    } catch (error) {
        console.error('Error fetching menu items:', error);
        return [];
    }
};

// GET /api/menu/item?id=1
export const getItemById = async (id) => {
    try {
        const response = await api.get(`/menu/item?id=${id}`);
        return response.data;
    } catch (error) {
        console.error('Error fetching menu item:', error);
        return null;
    }
};

// GET /api/menu/category?categoryId=1
export const getItemsByCategory = async (categoryId) => {
    try {
        const response = await api.get(`/menu/category?categoryId=${categoryId}`);
        return response.data;
    } catch (error) {
        console.error('Error fetching items by category:', error);
        return [];
    }
};

// POST /api/menu/add
export const addItem = async (itemData) => {
    try {
        const response = await api.post('/menu/add', itemData);
        return response.data;
    } catch (error) {
        console.error('Error adding menu item:', error);
        return { status: 'error', message: 'Failed to add item' };
    }
};

// PUT /api/menu/update
export const updateItem = async (itemData) => {
    try {
        const response = await api.put('/menu/update', itemData);
        return response.data;
    } catch (error) {
        console.error('Error updating menu item:', error);
        return { status: 'error', message: 'Failed to update item' };
    }
};

// PUT /api/menu/toggle?id=1&status=false
export const toggleAvailability = async (id, status) => {
    try {
        const response = await api.put(`/menu/toggle?id=${id}&status=${status}`);
        return response.data;
    } catch (error) {
        console.error('Error toggling availability:', error);
        return { status: 'error', message: 'Failed to toggle availability' };
    }
};

// DELETE /api/menu/delete?id=1
export const deleteItem = async (id) => {
    try {
        const response = await api.delete(`/menu/delete?id=${id}`);
        return response.data;
    } catch (error) {
        console.error('Error deleting menu item:', error);
        return { status: 'error', message: 'Failed to delete item' };
    }
};